Major version update for the Flare Sentinel Solution (3.0.0)

ireydiak opened the pull request:
**Summary:**
Major version update (v3.0.0) for the Flare Sentinel Solution. 
This release fixes Analytic Rules that fail when columns don't exist yet, introduces release notes to improve transparency around code changes, and replaces the deprecated Rest API connector with the new CFF connector. 

**Change(s):**
- Fixed Analytic Rules to use `column_ifexists()` for defensive querying
- Replaced Rest API connector with new CFF connector
- Added `ReleaseNotes.md` file

**Reason for Change(s):**
- Resolves errors when creating Analytic Rules before data is ingested
- Allows customers to enable rules immediately without waiting for data
- Improves communication with customers via release notes
- Follows maintenance requirements set by Azure

**Version updated:**
No

**Testing Completed:**
- Created a custom deployment using the "Deploy custom template" in Azure and pasted the `mainTemplate.yaml` content generated by the `createSolutionV3.ps1` utility script.
- Validated Analytic Rules syntax in Microsoft Sentinel environment.
- Confirmed Analytic Rules queries execute without errors.
- Confirmed Workbook creation execute without errors.

**Checked that the validations are passing and have addressed any issues that are present**:
- Yes

v-shukore commented:
Hi @ireydiak, 
We are putting your PR on hold because your solution uses Rest API connector and not Sentinel Codeless Connector Framework (CCF). We would be happy to work with you to migrate your solution to CCF.
Please drop a mail to the Microsoft Sentinel Partners at [AzureSentinelPartner@microsoft.com](mailto:AzureSentinelPartner@microsoft.com) to get their feedback and approval, so we can proceed further.
Thanks!!

ireydiak commented:
Hi @v-shukore. 
Sorry for the late reply. I was in vacation during the holiday season. We have another PR that uses the Codeless Connector. That PR is quite large and we wanted to split it in smaller ones (hence the current PR). I will create a new PR that introduces the Connector and you can tell me if we should split it or not.

ireydiak commented:
Hi @v-shukore 

- I modified `ReleaseNotes.md` to keep only the 3.0.0 version.
- I added instructions to remove empty arrays in the `mainTemplate.json` (there was no blank line so I think there might be a problem the validation script).
- I fixed the KQL queries.
- From what I understood, you want us to deleted the old connector as it is deprecated and won't be supported during the current year. As such I don't see a point in keeping this connector. Is there are reason we should keep it? 
- You will find the requested screenshots below.

### Connector
<img width="1567" height="849" alt="Screenshot 2026-01-12 at 10 12 09 AM" src="https://github.com/user-attachments/assets/8cda38f7-e759-4f2c-9f7f-2264c2f4cae6" />

### Workbook
<img width="1874" height="863" alt="Screenshot 2026-01-12 at 11 07 57 AM" src="https://github.com/user-attachments/assets/82723fd9-401e-48db-a4f7-55a49a59097f" />
<img width="1859" height="732" alt="Screenshot 2026-01-12 at 11 08 06 AM" src="https://github.com/user-attachments/assets/40dd89f1-4980-47c5-87bb-80c7d04caa20" />

v-shukore commented on Solutions/Flare/Data Connectors/FlareFireworkEventLogs_ccp/FlareFireworkEventLogs_DCR.json:
```diff
@@ -0,0 +1,167 @@
+    "name": "FireworkCustomDCR",
+    "apiVersion": "2021-09-01-preview",
+    "type": "Microsoft.Insights/dataCollectionRules",
+    "location": "[parameters('workspace-location')]",
```
please update here to "location": "{{location}}"

v-shukore commented:
Hi @ireydiak, please apply the same transformation for variable calls as well. You did it correctly in the pollingconfig file, so please make the same update for the other CCF files. Also, for this line: "location": "[parameters('workspace-location')]", update it accordingly. When all updates are done then repackage the solution once and commit the changes. Thanks!
